wikimedia/wikipedia-ios
# Allow for encoded urls

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T261027#6485517 and https://phabricator.wikimedia.org/T259902

### Notes
* In short, MediaWiki (or another part of the stack) ensures that links to Wikipedia articles are unencoded, but for off-wiki links (including ones to commons, for instance) the URL can be encoded or unencoded, depending on how the editor writes it. We now ensure we're not encoding already-encoded URLs, which was the problem leading to these bad links.
* Question mark: We also needed to allow the question mark to pass through unencoded, for query strings.
* And added tests for the range of these possible situations.

### Test Steps
**Tests**
Run the tests in `URLParsingAndRoutingTests`, ensure they pass.

**Manual testing**
1. Run the app. 
1. Navigate to the link mentioned in [this ticket](https://phabricator.wikimedia.org/T259902), ensure it now works.
1. Navigate to the link mentioned in the main part of [this ticket](https://phabricator.wikimedia.org/T261027), ensure it now works.

### Screenshots/Videos *(if applicable)*
